RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -335,6 +335,12 @@ class PumpOpsSynchronous {
 
         var events = [TimestampedHistoryEvent]()
         var timeAdjustmentInterval: NSTimeInterval = 0
+        
+        // Going to scan backwards in time through events, so event time should be monotonically decreasing
+        // Start with some time in the future, to account for the condition when the pump's clock is ahead
+        // of ours by a small amount.
+        var timeCursor = NSDate(timeIntervalSinceNow: NSTimeInterval(minutes: 60))
+        let eventTimestampDeltaAllowance = NSTimeInterval(minutes: 60)
```

Comment:
The only known mutable records, I believe, are Normal Bolus and Square
Bolus.

On Mon, Aug 29, 2016 at 12:53 PM Pete Schwamb <notifications@github.com>
wrote:

> In RileyLinkKit/PumpOpsSynchronous.swift
> <https://github.com/ps2/rileylink_ios/pull/211#discussion_r76674880>:
>
> > @@ -335,6 +335,12 @@ class PumpOpsSynchronous {
> >
> >          var events = [TimestampedHistoryEvent]()
> >          var timeAdjustmentInterval: NSTimeInterval = 0
> > +
> > +        // Going to scan backwards in time through events, so event time should be monotonically decreasing
> > +        // Start with some time in the future, to account for the condition when the pump's clock is ahead
> > +        // of ours by a small amount.
> > +        var timeCursor = NSDate(timeIntervalSinceNow: NSTimeInterval(minutes: 60))
> > +        let eventTimestampDeltaAllowance = NSTimeInterval(minutes: 60)
>
> Yes, it looks like you're right about newer models. Here's what I see on
> the 551:
>
> 1) Square wave boluses get added to history immediately upon start. The
> amount is 0.
> 2) Events can be added while square wave is delivering.
> 3) Square wave record in history does not update until delivery is
> finished (or pump is suspended). At that point, the amount is updated to
> reflect whatever amount has been delivered.
>
> So it looks like out of order history events might only be a problem for
> 822s. I'm wondering if there are any other records we may want to look
> into. If not, I can adjust eventTimestampDeltaAllowance based on pump
> model.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/ps2/rileylink_ios/pull/211/files/d9ac4190a2d925dbcf4a950ffa8113543ce79f82#r76674880>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAM_jAVylnyTuKu1M87tG8iVBruF-Tz8ks5qkzi4gaJpZM4Jvtxq>
> .
>